packages/lib/response/cache.ts

```diff
@@ -0,0 +1,35 @@
+export const getResponsesCacheTag = (surveyId: string) => `surveys-${surveyId}-responses`;
+
+export const getResponseCacheTag = (responseId: string) => `responses-${responseId}`;
+
+export const getEnvironmentResponsesCacheTag = (environmentId: string): string =>
+  `environments-${environmentId}-responses`;
+
+export const getResponsesByPersonIdCacheTags = (personId: string): string => `person-${personId}-responses`;
+
+/*
+ISSUES
+1. Pagination - how do we deal with findMany that are cached where page can be undefined
+2. How do we make sure tags are not duplicated in multiple functions
+
+3. How do we avoid writing long function names.
+The below proposal addresses this:
+
+serviceName + CacheTag
+
+Then every object within it begins with `by` and the dependency
+*/
+export const responseCacheTag = {
+  single(responseId: string) {
+    return `responses-${responseId}`;
+  },
+  bySurveyId(surveyId: string) {
+    return `surveys-${surveyId}-responses`;
+  },
+  byPersonId(personId: string) {
+    return `person-${personId}-responses`;
+  },
+  byEnvironmentId(environmentId: string) {
```

Comment:
I like the approach in general but I think it needs to be clear instantly if we revalidate only one response or multiple responses.

Reply:
my initial idea was for `single` to be responsible for one `response` and every other method in `responseCache` would be responsible for multiple responses.

What do you think?

Reply:
I think depending on our approach. If we decide for the function that we can pass different information I think I would prefer `byResponseId` instead of `single`. But except from this, like it

Reply:
yeah `byResponseId`, makes so much sense. yeah